### PR TITLE
BFD-1944: Refactor Locusts tests code to decrease complexity and remove redundant code

### PR DESCRIPTION
**Pull request by malessi**
**JIRA Ticket:**
[BFD-1944](https://jira.cms.gov/browse/BFD-1944)

**User Story or Bug Summary:**

As a BFD Engineer, I want the Locust tests project to be as free of unnecessary complexity as possible because this will make writing new, distinct test cases and suites against the BFD Server easier.

---

### What Does This PR Do?

This PR aims to improve the code quality of the Locusts tests through the following changes:
- Redundant "single-endpoint" Locustfiles under the `v1/` and `v2/` directories have all been removed.
  - Running tests against single or common endpoints has been preserved through the use of [`@tag`s](https://docs.locust.io/en/stable/writing-a-locustfile.html#tag-decorator). Each Locust task now has a set of associated tags that allow for users to filter which tasks to run, i.e.: `locust -f ... --tag=eob` or `locust -f ... --tag=patient_test_id`.
- Redundant and unnecessarily complex "abstract" `HttpUser` base classes like `BeneTestUser` or `ContractTestUser` have been out-right removed with their logic having been inlined into the relevant Locustfiles (`v*/regression_suite.py`, `v2/partially_adjudicated.py`, etc.).
  - These classes' existence seemed to be motivated by the existence of the single-endpoint Locustfiles which duplicated tasks from the regression suites. With the single-endpoint Locustfiles gone, these "abstract" classes/files have no real reason to persist.
  - Some of these classes created bad coupling between certain endpoint tasks that _appeared_ to be coupled, but were actually distinct endpoints. While it's not unlikely that these "pseudo-coupled" endpoints would continue to share the same invocation, the value gained from coupling them together was dubious at best and caused more friction in understanding what the tasks were _actually doing_  than necessary.
    - Considering the amount of code _removed_ by this PR, the amount of "duplicated" code added here is negligible in comparison (in fact, there is overall _less_ lines-of-code now even after formatting long lines). 
  - All of these classes' modules also loaded global data; this was causing some additional unnecessary complexity and, if persisted, would make implementing other forms of data loading (i.e. per-`HttpUser` data loading) confusing. Loading of data has also been "inlined" to the relevant Locustfiles, where it belongs.
  - The actual **functionality** of the Locust tasks defined in each of these "abstract" classes has been preserved and left unchanged in the Locustfiles they were "inlined" to (`v*/regression_suite.py`, ...).
- The opinionated Python [`black`](https://github.com/psf/black) formatter has been chosen as this sub-repository's default code formatter, and all Python files have been formatted with it in this PR.
  - The default line length was upped from `88` to `100`; no further changes to Black's default configuration were made.
- [`isort`](https://pycqa.github.io/isort/) has been ran using its built-in Black-compatible profile on each file to ensure that imports are sorted properly
- Pylint has been enabled for VSCode users for the workspace with some small tweaks to its default configuration via the included `.pylintrc`
- `settings.json` and `extensions.json` files have been added to this sub-repository under the `.vscode` directory to enforce some default workspace settings for VSCode users
  - As of this PR, the `settings.json` sets the following workspace-specific options:
    - The default Python formatting provider to `black`
    - Formatting runs on save, along with import organization
    - Import organization, using `isort`, runs with the `black`-compatible profile
    - Linting has been enabled, with the default linter being [`pylint`](https://pylint.pycqa.org/en/latest/)
      - Due to [issues with Pylint and VSCode resolving local module imports](https://github.com/Microsoft/python-language-server/issues/887), the `import-error` linting rule has been disabled by default _in VSCode only_. The `.pylintrc` does not specify disabling this linting error, so running `pylint` via the CLI will enforce that linting rule.
  - The `extensions.json` recommends installing some basic Python extensions, along with an `.editorconfig` extension
- An `.editorconfig` has been added for developers using editors other than VSCode (like emacs) to enforce a consistent Python code style.
- The remaining type hinting issues reported by Pyright/Pylance have all been resolved
- Validation logic has been corrected, with the default global failsafe check always checking for requests taking longer than `10000` ms. Additionally, V1 and V2 regression SLAs have been upped to a more reasonable value.
- The `locust` version requirement has been changed from `2.6.0` to the latest `2.10.1` version.
- Switch to using `FastHttpUser` instead of the normal `HttpUser` for `BFDUserBase` and all sub-classes of it.
  - Locust's [documentation on `FastHttpUser`](https://docs.locust.io/en/stable/increase-performance.html) claims that using `FastHttpUser` instead of the default `HttpUser` can increase the maximum number of requests "on a given hardware by as much as 5x-6x". Essentially, in constrained environments (like in a Jenkins job), this should help reduce the need for additional workers.
    - For example, running a simple suite against a [simple, locally hosted web server](https://hub.docker.com/r/strm/helloworld-http/) on an M1 Mac 2020 with 200 simulated users for 10 seconds resulted in `6985` total requests at `593.51` requests-per-second for the normal `HttpUser` and `8720` total requests at `736.74` requests-per-second for the `FastHttpUser`. Obviously not the 5x-6x claimed by Locust, but this test does show that `FastHttpUser` does have some performance benefits when resources are constrained.
  - `FastHttpUser` does have some caveats with respect to working with its "client library's internals", as it is based upon [`geventhttpclient`](https://github.com/geventhttpclient/geventhttpclient) instead of [`python-requests`](https://requests.readthedocs.io/en/latest/), but none of the tests in this project use such functionality and it is unlikely that any will need to in the future.

In short, a large amount of redundant code was removed. Unnecessarily coupled task code has been decoupled, and tasks have been inlined to the relevant Locustfiles. Data loading logic is now per-Locustfile. The code has been cleaned up, with various small issues and type hinting errors being fixed. The test's performance have also been potentially increased by up to 6x. And, finally, some developer QoL improvements have been made with choosing a default formatter, import sorter, and linter.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Do the changes make sense?
* Is the code in a good state to iterate on now?
  * Do we need any further changes to improve the quality of the code?  
* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/ccs-ops-misc/load_test/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "python.linting.pylintEnabled": true,
+  "python.linting.enabled": true,
+  // There appears to be issues with Pylint+VSCode resolving module paths for local
+  // directories, especially when using virutal environments. This is not
+  // disabled in .pylintrc, so running pylint manually will check imports.
+  // See https://github.com/Microsoft/vscode-python/issues/3840 and
+  // https://github.com/Microsoft/python-language-server/issues/887.
+  // None of the proposed solutions seemed to work, other than disabling this
+  // linting error
+  "python.linting.pylintArgs": ["--disable=import-error"]
```
Comment (mharvey) [inline ops/ccs-ops-misc/load_test/.vscode/settings.json:20]: I might have an idea for this, but it'll be out of band of this PR. Basically, I want to try defining a `.vscode-workspace` file _and_ a dev container for the workspace.

**Review by mharvey — APPROVED** [review]
Lots of improved developer comfort stuff here. 👍

**Review by mharvey — DISMISSED** [review]

**Review by mjburling — APPROVED** [review]
Rad.

**Review by mharvey — APPROVED** [review]

File: ops/ccs-ops-misc/load_test/.vscode/settings.json (hunk 1 above)
Comment (malessi) [reply ops/ccs-ops-misc/load_test/.vscode/settings.json:20]: Sounds good to me, I feel like we can definitely go down that avenue at some point in the future (maybe whenever we decide to include pre-commit hooks or move this into its own repo?).

**Review by KyRobbins — APPROVED** [review]

**Review by andrew-kramer — APPROVED** [review]
Looks great!

I wasn't able to run locally (due to my own accidental deletion of my test data), but I really like the direction and cleanliness of the code.

I understand why we had two different regression suites historically, but do we still need that? I _think_ the answer is probably still "yes", but wanted to make sure it was considered, at least.

**Comment by malessi** [comment]
> Looks great!
> 
> I wasn't able to run locally (due to my own accidental deletion of my test data), but I really like the direction and cleanliness of the code.
> 
> I understand why we had two different regression suites historically, but do we still need that? I _think_ the answer is probably still "yes", but wanted to make sure it was considered, at least.

@mharvey and I talked about this tangentially, but basically we were thinking that these suites are testing distinct endpoints and should probably be separate for the time being. We could probably come back to this and think about whether or not we should combine them at some point, or even if the current set of endpoint tests makes sense.

**Pull request merged**